src/ui/pages/Login/LayersAnimationLottie.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+} from 'react';
+import type { AnimationItem } from 'lottie-web';
+
+async function loadLottieWeb() {
+  const lottieModule = await import('lottie-web');
+  return lottieModule as unknown as (typeof lottieModule)['default'];
+}
+
+class LottieModule {
+  animationItem: AnimationItem | null = null;
+  options: { onAnimationReady?: () => void };
+  private disposables: Array<() => void> = [];
+
+  async prepareResources() {
+    const [lottieWeb, animationData] = await Promise.all([
+      loadLottieWeb(),
+      import('./login-animation-lottie.json'),
+    ]);
+    return { lottieWeb, animationData };
+  }
+
+  constructor({ onAnimationReady }: { onAnimationReady?: () => void } = {}) {
+    this.options = { onAnimationReady };
+  }
+
+  async startAnimation(
+    container: Element,
+    { autoplay, signal }: { autoplay: boolean; signal: AbortSignal }
+  ) {
+    const { lottieWeb, animationData } = await this.prepareResources();
+    if (signal.aborted) {
+      return;
+    }
+    const item = lottieWeb.loadAnimation({
+      container,
+      animationData,
+      loop: false,
+      autoplay,
+      renderer: 'svg',
+    });
+    function loopOnCompleted() {
+      const FRAME_NUMBER_TO_START_LOOP = 39;
```

Comment:
Probably the only line specific to this particular case. Everything else looks like can be reused easily.

Reply:
this and the path to the lottie file

Reply:
And the looping logic :)